src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -1356,3 +1358,33 @@ def should_encrypt_token_cache(cli_ctx):
     encrypt = cli_ctx.config.getboolean('core', 'encrypt_token_cache', fallback=fallback)
 
     return encrypt
+
+
+def subprocess_arg_type_check(args, kwargs):
+    skip_arg_type_check = kwargs.get("skip_arg_type_check", None)
+    if skip_arg_type_check is not None:
+        del kwargs["skip_arg_type_check"]
+    if skip_arg_type_check:
+        return
+    if isinstance(args, list):
+        return
+    from azure.cli.core.azclierror import ArgumentUsageError
+    raise ArgumentUsageError("cli_subprocess args should be a list of sequence")
+
+
+def subprocess_arg_mask(args, kwargs):
+    enable_arg_mask = kwargs.get("enable_arg_mask", None)
+    if enable_arg_mask is not None:
+        del kwargs["enable_arg_mask"]
+    if not enable_arg_mask:
+        return
+    if not isinstance(args, list):
+        return
+    for i, val in enumerate(args):
+        args[i] = re.sub(ARG_MASK_CHAR, "", val)
+
+
+def subprocess_kwarg_mask(kwargs):
+    if kwargs.get("shell", False):
+        logger.warning("Removed shell=True for cli processor")
+        kwargs["shell"] = False
```

Comment:
If we believe `shell=True` is unnecessary, why not raise an error to fail the tests and let the code owner to fix it?

Reply:
We can but that raises a developers' second-check issue, which actually can be easily eliminated. I prefer this to be a by-default behaviour for clisubprocess so that it is a one-time effort for developers to adopt it.

Reply:
I agree with @bebound that developers should be guided to follow the best practice, instead of silently changing the parameters.